mne_bids/write.py

```diff
@@ -478,13 +477,36 @@ def _scans_tsv(raw, raw_fname, fname, overwrite=False):
           for raw_f in raw_fnames]),
             ('acq_time', [acq_time] * len(raw_fnames))])
 
+    # add source filename if desired
+    if keep_source:
+        data['source'] = [Path(src_fname).name for src_fname in raw.filenames]
+
+        # write out a sidecar JSON if not exists
+        sidecar_json_path = Path(fname).with_suffix('.json')
+        sidecar_json_path = get_bids_path_from_fname(sidecar_json_path)
+        sidecar_json = [('source', 'Original source filename.')]
+        sidecar_json = OrderedDict(sidecar_json)
```

Comment:
I think this is still not right. The dict called `d` below is what you need:

```python
In [1]: import json

In [2]: d = dict(source=dict(Description="Original source filename."))

In [6]: print(json.dumps(d, indent=4))
{
    "source": {
        "Description": "Original source filename."
    }
}

```

Reply:
Why do you make use of an `OrderedDict` here? `dict` order is guaranteed to be in insertion order since Python 3.7

Reply:
Ah I see, yeah okay idk why I wasn't getting it.